### ISSUE TITLE
twisted.internet.test.test_posixbase.TCPPortTests.test_connectionLostFailed runs against nonsensical reactors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3677]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3677|
|Type|defect|
|Created|2009-03-02 17:18:51Z|
|Branch|https://github.com/twisted/twisted/tree/tcp-test-posixbase-3677|

If the test suite is run with a reactor not based on twisted.internet.posixbase, `twisted.internet.test.test_posixbase.TCPPortTests` still runs and uses that reactor.  This means the test might fail.  This is useless, since it's not intended to work on non-posixbase reactors.  It needs to be skipped or made to create its own posixbase-derived reactor instance.

<details><summary>Searchable metadata</summary>

```
trac-id__3677 3677
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_tcp_test_posixbase_3677 branches/tcp-test-posixbase-3677
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1236014331000000 1236014331000000
changetime__1244968569000000 1244968569000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [27006]) Branching to 'tcp-test-posixbase-3677'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve removed owner|
|-|-|

I added a simple skip, please review.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_4"></a>|@mwhudson set owner to @therve|
|-|-|

Looks fine to me, please land!

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set status to `closed`|
|-|-|

(In [27018]) Merge tcp-test-posixbase-3677

Author: therve
Reviewer: mwh
Fixes [#3677](https://github.com/twisted/twisted/issues/3677)

Skip tests expecting a PosixReactorBase variant on non-PosixReactorBase
reactors (ie iocp).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
